src/Uno.UWP/Storage/FileIO.cs

```diff
@@ -0,0 +1,224 @@
+				{
+					try
+					{
+						byte firstByte = binReader.ReadByte();
+						switch (firstByte)
+						{
+							case 0xfe:
+								unicodeEncoding = Streams.UnicodeEncoding.Utf16BE;
+								break;
+							case 0xff:
+								unicodeEncoding = Streams.UnicodeEncoding.Utf16LE;
+								break;
+						}
+					}
+					catch
+					{
+						// probably file is empty
+					}
+
+				}
+			}
+
+			AppendWriteTextAsync(file, contents, unicodeEncoding, append);
+		}
+
+		public static IAsyncAction WriteTextAsync(IStorageFile file, string contents) => AppendWriteTextAsync(file, contents, false).AsAsyncAction();
+
+		public static IAsyncAction AppendTextAsync(IStorageFile file, string contents) => AppendWriteTextAsync(file, contents, true).AsAsyncAction();
+
+		public static IAsyncAction WriteLinesAsync(IStorageFile file, IEnumerable<string> lines)
+			=> WriteTextAsync(file, ConvertLinesToString(lines));
+		public static IAsyncAction AppendLinesAsync(IStorageFile file, IEnumerable<string> lines)
+			=> AppendTextAsync(file, ConvertLinesToString(lines));
+
+
+		private static async Task<string> ReadTextTaskAsync(IStorageFile file, Streams.UnicodeEncoding encoding)
+		{
+			if (file is null)
+			{
+				// UWP throws NullReferenceException
+				// but it was choosen to throw ArgumentNullException , as it is more appriopriate in this caase
+				throw new ArgumentNullException("StorageFile cannot be null");
+			}
+
+			Encoding encodingForReader = Encoding.UTF8;
+			switch (encoding)
+			{
+				case Streams.UnicodeEncoding.Utf8:
+					encodingForReader = Encoding.UTF8;
+					break;
+				case Streams.UnicodeEncoding.Utf16LE:
+					encodingForReader = Encoding.Unicode;
+					break;
+				case Streams.UnicodeEncoding.Utf16BE:
+					encodingForReader = Encoding.BigEndianUnicode;
+					break;
+			}
+
+			string output = "";
+			using (Stream fileStream = await file.OpenStreamForReadAsync())
+			{
+				using (StreamReader streamReader = new StreamReader(fileStream, encodingForReader))
+				{
+					output = streamReader.ReadToEnd();
+				}
+			}
+			return output;
+		}
+
+		public static IAsyncOperation<string> ReadTextAsync(IStorageFile file, Streams.UnicodeEncoding encoding)
+			=> ReadTextTaskAsync(file, encoding).AsAsyncOperation<string>();
+
+		private static async Task<string> ReadTextTaskAsync(IStorageFile file)
+		{
+			if (file is null)
+			{
+				// UWP throws NullReferenceException
+				// but it was choosen to throw ArgumentNullException , as it is more appriopriate in this caase
+				throw new ArgumentNullException("StorageFile cannot be null");
+			}
+
+			string output = "";
+			Stream fileStream = null;
+			try
+            {
+				fileStream = await file.OpenStreamForReadAsync();
+				using (StreamReader streamReader = new StreamReader(fileStream))
+				{
+					output = streamReader.ReadToEnd();
+				}
+			}
+			finally
+			{
+				if(fileStream != null)
+                {
+					fileStream.Dispose();

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to make sure 'fileStream' is disposed only once.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4827590)

Reply:
This is already done, changed from using/using to try/using/finally. I don't understand why codacy still says that it is error.

Reply:
It's already disposed by the `using` on the `StreamReader`.

https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.-ctor?view=netframework-4.8#System_IO_StreamReader__ctor_System_IO_Stream_System_Text_Encoding_System_Boolean_System_Int32_System_Boolean_
> Unless you set the leaveOpen parameter to true, the StreamReader object calls Dispose() on the provided Stream object when StreamReader.Dispose is called.

Reply:
I used `using(fileStream.. { using(streamWriter... } }` originally.
Codacy suggested same thing as now - change code to use `try { ... using(streamWriter...} finally` (see link from codacy, `Compliant Solution` section). So, I changed code.

And even if .Net API says that StreamReader disposes Stream, it cost nothing to try to stream.Dispose (), and it would be protection for bad implementation of StreamReader.Dispose(). I think calling dispose twice is better than memory/resources leak.

Reply:
I agree. Codacy is _smart enough_ to know the `StreamWriter` will dispose the stream, but as you said in this case it's not a problem.

But it's common for components to throw an `ObjectDisposedException` when trying to dispose twice the same object. That's why this rule exists.